src/org/parosproxy/paros/core/scanner/VariantAbstractQuery.java

```diff
@@ -123,6 +124,10 @@ protected void setParameters(int type, List<org.zaproxy.zap.model.NameValuePair>
             listParam.add(new NameValuePair(type, nonNullString(parameter.getName()), nonNullString(parameter.getValue()), i));
             i++;
         }
+        if (i == 0) {
+        	// No query params, lets add one just to make sure
+            listParam.add(new NameValuePair(type, "query", "query", i));

```

Comment:
It could be, although I'm wary of adding options that I dont think people will use ;)
I dont think this is radically different to what we already do, so I cant see why someone would want to turn it off.
Any other views?